editor/plugins/mesh_instance_3d_editor_plugin.cpp

```diff
@@ -343,6 +343,55 @@ void MeshInstance3DEditor::_menu_option(int p_option) {
 			}
 			_create_uv_lines(1);
 		} break;
+		case MENU_OPTION_CREATE_TRAIL_SKELETON: {
+			if (node == get_tree()->get_edited_scene_root()) {
+				err_dialog->set_text(TTR("MeshInstance3D cannot be the scene's root node."));
+				err_dialog->popup_centered();
+				return;
+			}
+
+			if (!node->get_mesh()->is_class("TubeTrailMesh") && !node->get_mesh()->is_class("RibbonTrailMesh")) {
+				err_dialog->set_text(TTR("Mesh must be a TubeTrailMesh or a RibbonTrailMesh."));
+				err_dialog->popup_centered();
+				return;
+			}
+
+			Skeleton3D *skeleton = Object::cast_to<Skeleton3D>(node->get_node(node->get_skeleton_path()));
+
```

Comment:
Can the skeleton be null? Can use a ERR_FAIL_NULL

Reply:
Actually that is a mistake, the skeleton is supposed to be created new every time, I asked and no one had any idea what the behaviour should be if the skeleton already exists.